pkg/controller/migplan/validation.go

```diff
@@ -581,6 +592,83 @@ func (r ReconcileMigPlan) validateConflict(plan *migapi.MigPlan) error {
 	return nil
 }
 
+// Validiate Namespace Selection
+func (r ReconcileMigPlan) validateNamespaceSelection(plan *migapi.MigPlan) error {
+	invalidNamespaces := make([]string, 0)
+
+	if plan.Status.HasAnyCondition(Suspended) {
+		return nil
+	}
+
+	cluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	identity, err := plan.GetSourceIdentity(client)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	for i := 0; i < len(plan.Spec.Namespaces); i++ {
+		access, err := canUserAccess(client, identity, plan.Spec.Namespaces[i])
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+		if !access {
+			invalidNamespaces = append(invalidNamespaces, plan.Spec.Namespaces[i])
+			continue
+		}
+	}
+	if len(invalidNamespaces) > 0 {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     InvalidNamespaceSelection,
+			Status:   True,
+			Reason:   NotDone,
+			Category: Error,
+			Message:  InvalidNamespaceSelectionMessage,
+			Items:    invalidNamespaces,
+		})
+		return errors.New("selected namespaces are invalid")
+	}
+
+	return nil
+}
+
+func canUserAccess(client k8sclient.Client, user, namespace string) (bool, error) {
```

Comment:
I'm wondering if putting this, and what the migration controller will need to do in a new top level `auth` package.
```
package auth

type Authorized map[string]bool

func HasRead(token string, namespaces []string) (Authorized, error)

func HasMigrate(token string, namespace []string) (Authorized, error)
```
For these reasons:
- The HasRead can be shared by the discovery controller.
- In both cases, a new client needs to be constructed and can be reused to check each namespace.
- Centralizes optimizations.
- Seems helpful to have one place to maintain how auth is done.